hack/tools/generate-yaml/Dockerfile

```diff
@@ -49,6 +49,9 @@ COPY ./vendor/sigs.k8s.io/cluster-api/config ./vendor/sigs.k8s.io/cluster-api/co
 # Copy in the examples directory.
 COPY ./cmd/clusterctl/examples/vsphere/*.template ./cmd/clusterctl/examples/vsphere/
 
+# Ensure all the directories in the /build directory are writeable.
+RUN find . -type d -exec chmod 0777 \{\} \;
```

Comment:
Maybe I'm missing something, but this updates `/build` to have 0777 but the yaml files are still generated under `/out`.

Reply:
Because before the user couldn't write the generated manager image patch into the directory since they were `root:root 0755`. This allows the mapped in `user:group` to write to `/out`

Reply:
But `/out` is still not accessible I think. If you mount the parent directory and let `generate-yamal.sh` create the directory it works:
```
docker run --rm   --user "$(id -u):$(id -g)" -e OUT_DIR=/build/capv/out/   -v "$(pwd)/":/build/capv/   -v "$(pwd)/envvars.txt":/out/envvars.txt:ro   gcr.io/cluster-api-provider-vsphere/release/manifests:latest   -c management-cluster
```

But as long as `/out` is being mounted directly it is owned by root (I think).

Reply:
*sigh*

For those watching this, it turns out this behaves differently whether you're running Docker on Mac or Linux. Fun.